### PR TITLE
fix: use the real path to node_modules

### PR DESCRIPTION
**Pull request by etamponi**

### CODE REVIEW

File: src/api/install.ts [hunk 1]
```diff
@@ -325,7 +317,10 @@ async function installInContext (
     logger.warn('`node_modules` is present. Shrinkwrap only installation will make it out-of-date')
   }
 
-  const nodeModulesPath = path.join(ctx.root, 'node_modules')
+  let nodeModulesPath = path.join(ctx.root, 'node_modules')
+  if (await fs.exists(nodeModulesPath)) {
+    nodeModulesPath = await fs.realpath(nodeModulesPath)
```
Comment (zkochan) [inline src/api/install.ts]: the correct way to do it is put `fs.realpath` into a try/catch and on ENOENT use just `path.join`

```ts
function safeRealpath (filename: string) {
  try {
    return await fs.realpath(filename)
  } catch (err) {
    if (err['code'] === 'ENOENT') return filename
    throw err
  }
}
```

**Comment by zkochan** [comment]
Add one test with a symlinked `node_modules`

File: src/api/install.ts (hunk 1 above)
Comment (zkochan) [reply src/api/install.ts]: the reason for that is because between calls to exists and realpath the directory could've been removed

**Comment by etamponi** [comment]
What's the behaviour to test?

UPDATE: I added a test for local packages and symlinked node_modules

File: src/fs/realNodeModulesDir.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import fs = require('mz/fs')
+import path = require('path')
+
+const CACHE: {[prefix: string]: string} = {}
+
+export default async function realNodeModulesDir (prefix: string): Promise<string> {
```
Comment (zkochan) [inline src/fs/realNodeModulesDir.ts]: would it be possible to make this function stateless? Maybe the realNodeModules could be passed along with prefix in opts/ctx

**Review by zkochan — APPROVED** [review]
LGTM

**Comment by zkochan** [comment]
Oh, you didn't update the link() API description in the README. I'll do it https://github.com/pnpm/supi#supilinklinkfrom-lintto-options

**Pull request merged**